numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1507,7 +1507,8 @@ arr_add_docstring(PyObject *NPY_UNUSED(dummy), PyObject *args)
         PyObject *doc_attr;
 
         doc_attr = PyObject_GetAttrString(obj, "__doc__");
-        if (doc_attr != NULL && doc_attr != Py_None) {
+        if (doc_attr != NULL && doc_attr != Py_None &&
+                (PyUnicode_Compare(doc_attr, obj) != 0)) {
             PyErr_Format(PyExc_RuntimeError, "object %s", msg);
```

Comment:
```suggestion
            if (PyErr_Occurred()) {
                return NULL;  // PyUnicode_Compare can fail
            }
            PyErr_Format(PyExc_RuntimeError, "object %s", msg);
```

Reply:
Right, seems there is also a %R or so missing to tell you which object we are talking about...

Reply:
Changed, the check was actually incorrect, it should have been `str` not `obj`, its fixed now (tested manually with `np.add_newdoc(np.take_along_axis, "wrong docstring")`)

BUT: We currently expose even `np.add_newdoc` as top-level... So I reverted the try/except change there, since it could be very disruptive in theory (i.e. something not importing). My tendency is to give a warning instead as soon as 1.20 is out.